### PR TITLE
Clean up changes to add telephone field

### PR DESCRIPTION
**Pull request by chaselivingston**

### CODE REVIEW

**Comment by chaselivingston** [comment]
Related #955

**Comment by samhotchkiss** [comment]
This is failing in travis-- let's punt it to 3.5

**Comment by chaselivingston** [comment]
Any idea what's causing it to fail? I'm not super familiar with Travis.

**Comment by samhotchkiss** [comment]
https://travis-ci.org/Automattic/jetpack/builds/48657267

`Parse error: syntax error, unexpected 'break' (T_BREAK) in /tmp/wordpress/src/wp-content/plugins/jetpack/modules/contact-form/grunion-contact-form.php on line 1678`

**Comment by chaselivingston** [comment]
Fixed @samhotchkiss, thanks for the heads up!

File: modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -1252,6 +1253,7 @@ function process_submission() {
 
 		// Initialize all these "standard" fields to null
 		$comment_author_email = $comment_author_email_label = // v
+		$comment_author_telephone = $comment_author_telephone_label = // v
```
Comment (georgestephanis) [inline modules/contact-form/grunion-contact-form.php]: These are only for standard comment fields, no?  Telephone isn't a standard comment field ...

File: modules/contact-form/grunion-contact-form.php [hunk 2]
```diff
@@ -1668,9 +1670,12 @@ function render() {
 		) {
 			// Special defaults for logged-in users
 			switch ( $this->get_attribute( 'type' ) ) {
-			case 'email';
+			case 'email' :
 				$this->value = $current_user->data->user_email;
 				break;
+			case 'telephone' :
+				$this->value = $current_user->data->user_telephone;
```
Comment (georgestephanis) [inline modules/contact-form/grunion-contact-form.php]: There is no standard `user_telephone` usermeta -- where is this coming from?

File: modules/contact-form/grunion-contact-form.php [hunk 3]
```diff
@@ -1691,9 +1696,13 @@ function render() {
 		case 'email' :
 			$r .= "\n<div>\n";
 			$r .= "\t\t<label for='" . esc_attr( $field_id ) . "' class='grunion-field-label email" . ( $this->is_error() ? ' form-error' : '' ) . "'>" . esc_html( $field_label ) . ( $field_required ? '<span>' . __( "(required)", 'jetpack' ) . '</span>' : '' ) . "</label>\n";
-			$r .= "\t\t<input type='email' name='" . esc_attr( $field_id ) . "' id='" . esc_attr( $field_id ) . "' value='" . esc_attr( $field_value ) . "' class='email' " . $field_placeholder . " " . ( $field_required ? "required aria-required='true'" : "" ) . "/>\n";
+			$r .= "\t\t<input type='email' name='" . esc_attr( $field_id ) . "' id='" . esc_attr( $field_id ) . "' value='" . esc_attr( $field_value ) . "' class='telephone' " . $field_placeholder . " " . ( $field_required ? "required aria-required='true'" : "" ) . "/>\n";
```
Comment (georgestephanis) [inline modules/contact-form/grunion-contact-form.php]: Why did this change from `class="email"` to `class="telephone"` ?

File: modules/contact-form/grunion-contact-form.php [hunk 4]
```diff
@@ -1691,9 +1696,13 @@ function render() {
...
 			$r .= "\t</div>\n";
 			break;
+		case 'telephone' :
+			$r .= "\n<div>\n";
+			$r .= "\t\t<label for='" . esc_attr( $field_id ) . "' class='grunion-field-label telephone" . ( $this->is_error() ? ' form-error' : '' ) . "'>" . esc_html( $field_label ) . ( $field_required ? '<span>' . __( "(required)", 'jetpack' ) . '</span>' : '' ) . "</label>\n";
+			$r .= "\t\t<input type='tel' name='" . esc_attr( $field_id ) . "' id='" . esc_attr( $field_id ) . "' value='" . esc_attr( $field_value ) . "' class='email' " . $field_placeholder . "/>\n";
```
Comment (georgestephanis) [inline modules/contact-form/grunion-contact-form.php]: Why does this have `class="email"` for a telephone attribute?

**Pull request merged**